modules/edw_event/modules/edw_event_agenda/src/Services/MeetingAgendaService.php

```diff
@@ -113,4 +111,48 @@ public function loadMultipleAgendaItems(array $ids) {
     return $this->termStorage->loadMultiple($ids);
   }
 
+  /**
+   * Creates default agenda for a meeting.
+   *
+   * @param int $meetingId
+   *   The meeting id.
+   * @param string $agendaName
+   *   The agenda's name.
+   *
+   * @throws \Drupal\Core\Entity\EntityStorageException
+   */
+  public function createDefaultAgenda(int $meetingId, string $agendaName) {
+    $defaultAgenda = $this->termStorage->getQuery()->accessCheck(FALSE)
+      ->condition('vid', 'event_agendas')
+      ->condition('field_event', $meetingId)
+      ->condition('field_is_default_agenda', TRUE)
+      ->execute();
+    $defaultAgenda = reset($defaultAgenda);
+
+    if (!empty($defaultAgenda)) {
+      return;
+    }
+
+    // If there is a non-default agenda with that name make it default.
+    $properties = [
+      'vid' => 'event_agendas',
+      'field_event' => $meetingId,
+      'name' => $agendaName,
+    ];
+    $term = $this->termStorage->loadByProperties($properties);
+    $term = reset($term);
+
+    if (!empty($term)) {
+      $term->set('field_is_default_agenda', TRUE);
+      $term->save();
+      return;
+    }
+
+
+    // Otherwise, create a new default agenda.
+    $properties['field_is_default_agenda'] = TRUE;
+    $defaultAgenda = $this->termStorage->create($properties);
+    $defaultAgenda->save();
```

Comment:
Display a message when the term is created. (e.g.: A new agenda item with id x was created.)